letsencrypt/client.py

```diff
@@ -105,7 +105,8 @@ def register(config, account_storage, tos_cb=None):
                    "--register-unsafely-without-email was not present.")
             logger.warn(msg)
             raise errors.Error(msg)
-        logger.warn("Registering without email!")
+        if not config.dry_run:
```

Comment:
> Should we also suppress this message in staging mode? Or do we expect people to know about their staging accounts?

I don't have a strong opinion on this either way. If we want to suppress this message when using the staging server, we can simply change this line to `if not config.dry_run` (and fix the test that will break).

> If not, should we note when what we're registering for is a staging account?

I don't think so. Treating servers differently in places other than `cli.py` seems clunky and unnecessary. The only way we're going to ask a user to register for a staging account is if they specifically requested it with the CLI/config file. Given this, adding a special prompt here doesn't seem like a big gain.